packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -209,7 +210,18 @@ public void onPlaybackStateChanged(final int playbackState) {
           @Override
           public void onPlayerError(final ExoPlaybackException error) {
             setBuffering(false);
-            if (eventSink != null) {
+            if (isBehindLiveWindow(error) && exoPlayer != null) {
```

Comment:
All of the changes in this file--which is the core of the PR--could be reverted without any tests failing, so the PR doesn't have meaningful test coverage. This needs a native unit test.